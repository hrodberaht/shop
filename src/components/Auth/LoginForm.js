import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getAuthError } from '../../store/auth/selectors';
import { login } from '../../store/auth/actionCreator';

export class LoginForm extends Component {
  state = {
    email: '',
    password: '',
  };

  handleChange = (e) => {
    this.setState({ [e.target.id]: e.target.value });
  };

  resetState = () => {
    this.setState({
      email: '',
      password: '',
    });
  };

  handleSubmit = (e) => {
    const { email, password } = this.state;
    const { handleLogin } = this.props;
    e.preventDefault();
    this.resetState();
    handleLogin(email, password);
  };

  render() {
    const { email, password } = this.state;
    const { error } = this.props;
    return (
      <form onSubmit={this.handleSubmit}>
        <label className="form-label" htmlFor="email">
          <p>Email:</p>
          <input
            className="form-input"
            id="email"
            type="text"
            value={email}
            onChange={this.handleChange}
            required
          />
        </label>
        <label className="form-label" htmlFor="password">
          <p>Password:</p>
          <input
            className="form-input"
            id="password"
            type="password"
            value={password}
            onChange={this.handleChange}
            required
          />
        </label>
        <button type="submit">Log In</button>
        <p className="error-text">{error}</p>
      </form>
    );
  }
}

const mapStateToProps = state => ({
  error: getAuthError(state),
});

export default connect(
  mapStateToProps,
  { handleLogin: login },
)(LoginForm);

LoginForm.propTypes = {
  handleLogin: PropTypes.func.isRequired,
  error: PropTypes.string,
};

LoginForm.defaultProps = {
  error: null,
};
