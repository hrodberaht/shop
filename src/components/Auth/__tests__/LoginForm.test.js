import { LoginForm } from '../LoginForm';

const defaultProps = {
  handleLogin: jest.fn(),
};
const setup = buildSetup(LoginForm, defaultProps);

const setValuesToInputs = (wrapper) => {
  wrapper.find('#email').simulate('change', { target: { value: 'test@test.pl', id: 'email' } });
  wrapper.find('#password').simulate('change', { target: { value: '1234', id: 'password' } });
};

describe('<LoginForm />', () => {
  it('should render witout crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleLogin with parameters email and password after submit', () => {
    const { wrapper } = setup();
    setValuesToInputs(wrapper);
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(wrapper.instance().props.handleLogin).toHaveBeenCalledWith('test@test.pl', '1234');
  });

  it('after set value to input the state should change', () => {
    const { wrapper } = setup();
    setValuesToInputs(wrapper);
    expect(wrapper.state()).toEqual({ email: 'test@test.pl', password: '1234' });
  });

  it('should clear state after submit', () => {
    const { wrapper } = setup();
    setValuesToInputs(wrapper);
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(wrapper.state()).toEqual({ email: '', password: '' });
  });
});
