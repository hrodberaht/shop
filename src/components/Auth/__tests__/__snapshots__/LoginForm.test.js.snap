// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render witout crash 1`] = `
<div>
  <div
    className="login-form"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        <p>
          Email:
        </p>
        <input
          className="form-input"
          id="email"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
      </label>
      <label
        className="form-label"
        htmlFor="password"
      >
        <p>
          Password:
        </p>
        <input
          className="form-input"
          id="password"
          onChange={[Function]}
          required={true}
          type="password"
          value=""
        />
      </label>
      <button
        className="btn btn-submit"
        type="submit"
      >
        Log In
      </button>
      <p
        className="error-text"
      />
    </form>
  </div>
  <div
    className="registartion-link"
  >
    <Link
      replace={false}
      to="/registration"
    >
      Registration
    </Link>
  </div>
</div>
`;
